### PR TITLE
[IE VPU] switch unroll loops to ngraph

### PR DESCRIPTION
**Pull request by KorA6**
### Details:
 - *Switch VPU unroll loops to ngraph*

### Tickets:
 - *52923*

### CODE REVIEW

**Review by DariaMityagina — COMMENTED** [review]
In general, looks OK to me.
But is it still WIP? If not, please remove this label

**Comment by KorA6** [comment]
> In general, looks OK to me.
> But is it still WIP? If not, please remove this label

The label has been removed

**Review by DariaMityagina — APPROVED** [review]

File: inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp [hunk 1]
```diff
@@ -219,7 +227,44 @@ ie::CNNNetwork FrontEnd::convertNetwork(ie::CNNNetwork& network) {
...
+                              ngraph::pass::ConvertTensorIteratorToLSTMSequence,
+                              ngraph::pass::ConvertTensorIteratorToGRUSequence>(
+            [&env, isCellPrimitiveSupported](const_node_ptr &node) -> bool {
+                if (env.config.forcePureTensorIterator) {
+                    return false;
+                }
+                if (const auto& ti_op = std::dynamic_pointer_cast<const ngraph::op::TensorIterator>(node)) {
+                    size_t count_rnn = 0;
+                    for (const auto &op : ti_op->get_body()->get_ops())
+                        count_rnn += isCellPrimitiveSupported(op);
```
Comment (gladilov-gleb) [inline inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:259]: Nitpick; you can use `std::count_if`

File: inference-engine/src/vpu/graph_transformer/src/stages/rnn.cpp [hunk 2]
```diff
@@ -199,8 +199,16 @@ void FrontEnd::parseRNN(const Model& model, const ie::CNNLayerPtr& _layer, const
...
+    const auto& validOutDataIter = std::find_if(outputData.begin(), outputData.end(), [](const vpu::Data& data) {
+        return data != nullptr;
+    });
+
+    VPU_THROW_UNLESS(validOutDataIter != outputData.end(), "Layer {} with type {} failed: all outputs is nullptr", layer->name, layer->type);
+
+    for (auto& out : outputData) {
+        if (out == nullptr) {
+            out = model->addFakeData();
+        }
```
Comment (gladilov-gleb) [inline inference-engine/src/vpu/graph_transformer/src/stages/rnn.cpp:211]: Nitpick: you can use `std::transform` + `std::back_inserter`

**Review by gladilov-gleb — APPROVED** [review]
LGTM, minor comments

**Pull request closed without merging**